### PR TITLE
Add openebs-mixin

### PR DESCRIPTION
**Pull request by rajaSahil**
Signed-off-by: Sahil Raja <sahilraja242@gmail.com>

### Reviewer Notes

- Please ignore the changes in [monitoring/jsonnet/manifests/](https://github.com/rajaSahil/monitoring/tree/Add-Openebs-Mixin/jsonnet/manifests) directory. These are the files generated through the help of jsonnet and gojsontoyaml.

- Please ignore the changes in [monitoring/deploy/charts/openebs-monitoring/dashboards](https://github.com/rajaSahil/monitoring/tree/Add-Openebs-Mixin/deploy/charts/openebs-monitoring/dashboards) and [monitoring/deploy/charts/openebs-monitoring/rules](https://github.com/rajaSahil/monitoring/tree/Add-Openebs-Mixin/deploy/charts/openebs-monitoring/rules) as these are generated using jsonnet. Commit: 6ef162eba591f54ebc5442232a64aaa84390e9b0

- All the dashboards json present in [monitoring/jsonnet/openebs-mixin/dashboards](https://github.com/rajaSahil/monitoring/tree/Add-Openebs-Mixin/jsonnet/openebs-mixin/dashboards) are same as that of [dashboards](https://github.com/openebs/monitoring/tree/develop/deploy/charts/openebs-monitoring/dashboards).

- Alert rules defined in [monitoring/jsonnet/openebs-mixin/rules/](https://github.com/rajaSahil/monitoring/tree/Add-Openebs-Mixin/jsonnet/openebs-mixin/rules) are taken from [rules](https://github.com/openebs/monitoring/tree/develop/deploy/charts/openebs-monitoring/rules)

### What this PR does?

- Add openebs-mixin
- Add openebs monitoring using jsonnet
- Generate dashboards and alert rules for helm charts through openebs-mixin.

#### openebs-mixin
- Consists of all the dashboards and alert rules for openebs.

#### openebs monitoring using jsonnet
- openebs monitoring using jsonnet which has an external dependency of kube-Prometheus. It uses openebs-mixin for dashboards and alert rules.

#### Directory structure:
```
openebs/monitoring/jsonnet
├── addons # openebs-monitoring addon
│   ├── dashboard-cm.libsonnet # dashboard configmap to include openebs-* dashboard json files from openebs-mixin
│   ├── podMonitors.libsonnet # podMonitor to be created for different cas types
│   ├── prometheusRules.libsonnet # create prometheusRule object which includes alert rules defined in openebs-mixin
│   └── serviceMonitors.libsonnet # serviceMonitor to be created for different cas types
├── buildscripts # scripts and code required to build the project
│   ├── build.sh # script to generate manifests
│   ├── go.mod
│   ├── go.sum
│   └── tools.go
├── config.libsonnet # configuration to be used for openebs monitoing
├── jsonnetfile.json # direct dependencies
├── jsonnetfile.lock.json # all dependencies with exact version
├── lib # contains code that is used in different places
│   └── utils.libsonnet # code which is used to create openebs-monitoring addons
├── main.jsonnet # entrypoint of code
├── Makefile
├── manifests # generated yamls
│   ├── openebs-addons # openebs-monitoring addons yaml
│   │   ├── openebs-*.yaml
│   ├── setup # kube-prometheus crds, rbac and prometheus-operator yaml
│   │   ├── ...
│   ├── ... # kube-prometheus components yaml
└── openebs-mixin # openebs-mixin for dashboards and alert rules
    ├── build.sh
    ├── config.libsonnet # configuration that is used while writing dashboards and alert rules
    ├── dashboards
    │   ├── cstor-overview.json
    │   ├── cstor-pool.json
    │   ├── cstor-volume.json
    │   ├── cstor-volume-replica.json
    │   ├── dashboards.libsonnet # to import all the dashboard json
    │   ├── jiva-volume.json
    │   └── localpv-workload.json
    ├── jsonnetfile.json # direct dependencies
    ├── lib
    │   ├── dashboards.jsonnet
    │   └── rules.jsonnet
    ├── Makefile
    ├── mixin.libsonnet # openebs-mixin that imports all the dashboards and alert rules
    └── rules # contains alert rules for different cas types
        ├── cstor-pool-rules.libsonnet
        ├── cstor-rules.libsonnet
        ├── cstor-volume-rules.libsonnet
        ├── jiva-rules.libsonnet
        ├── jiva-volume-rules.libsonnet
        ├── rules.libsonnet
        └── volume-rules.libsonnet
```

#### Note: 
 Refer to this document for more insight into openebs-mixin and openebs-monitoring using jsonnet: 

- https://docs.google.com/document/d/1xJAjByPysQUHkxGK8gNUAALTt3f-eggGdLpNX4PiXn4/edit?usp=sharing
- https://docs.google.com/document/d/10SDX5dAQIoM5C3TP4QM4RibefVpSPXDy7a-NNUohjX0/edit?usp=sharing

### CODE REVIEW

File: deploy/charts/openebs-monitoring/templates/prometheusRules.yaml [hunk 1]
```diff
@@ -5,12 +5,12 @@
 {{ $_ := set $customEnabledRules $dirName $fields.enabled }}
 {{- end }}
 {{- end }}
-{{- $files := .Files.Glob "rules/**.yaml" }}
+{{- $files := .Files.Glob "rules/**.json" }}
```
Comment (rajaSahil) [inline deploy/charts/openebs-monitoring/templates/prometheusRules.yaml:8]: Alert rules will be generated in json using jsonnet.

File: deploy/charts/openebs-monitoring/values.yaml [hunk 2]
```diff
@@ -178,7 +178,7 @@ alertRules:
     enabled: true
   Jiva:
     enabled: true
-  volumes:
+  Volume:
```
Comment (rajaSahil) [inline deploy/charts/openebs-monitoring/values.yaml:204]: volume-rules.json is present in `rules/Volume/`.

File: jsonnet/buildscripts/build.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# Make sure to start with a clean 'manifests' dir
+rm -rf manifests
+mkdir -p manifests/setup
+mkdir -p manifests/openebs-addons
+
+# Calling gojsontoyaml is optional, but we would like to generate yaml, not json
+jsonnet -J vendor -m manifests "${1-main.jsonnet}" | xargs -I{} sh -c 'cat {} | gojsontoyaml > {}.yaml' -- {}
+
+# Make sure to remove json files
+find manifests -type f ! -name '*.yaml' -delete
```
Comment (avishnu) [inline jsonnet/buildscripts/build.sh:22]: what is ! for?

File: jsonnet/config.libsonnet [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    openebsMonitoring: {
+      namespace: 'openebs-monitoring',
+    },
+    // Configuration for different cas types.
+    casType: {
+      cStor: {
+        // To generate manifests for cstor. If set, manifests will be generated for cStor.
+        enabled: true,
+        // To generate dashboards configMap yamls. If set, dashboards will be appended in grafana-dashboardDefinition yaml.
+        installDashboards: true,
```
Comment (avishnu) [inline jsonnet/config.libsonnet]: should this be just dashboards as below there is just alertRules?

File: jsonnet/config.libsonnet [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+              },
+            ],
+          },
+        },
+        // PodMonitor configuration
+        podMonitor: {
+          enabled: false,
+        },
+      },
+      lvmLocalPV: {
```
Comment (avishnu) [inline jsonnet/config.libsonnet:220]: Can we add one for device localpv as well?

File: jsonnet/jsonnetfile.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (avishnu) [inline jsonnet/jsonnetfile.json:1]: Is the filename as per the standard or can it be dependency-jsonnet.json?

File: jsonnet/openebs-mixin/rules/prometheusRules.libsonnet [hunk 7]
```diff
@@ -0,0 +1,11 @@
+local cstor = (import 'cstor-rules.libsonnet');
```
Comment (avishnu) [inline jsonnet/openebs-mixin/rules/prometheusRules.libsonnet]: Bit inconsistent on naming the file, cstor-rules and jiva-rules, but prometheusRules.

File: jsonnet/buildscripts/build.sh (hunk 3 above)
Comment (avishnu) [reply jsonnet/buildscripts/build.sh:22]: Is it to exclude yaml files?

Comment (rajaSahil) [reply jsonnet/buildscripts/build.sh:22]: > Is it to exclude yaml files?

Yes

File: jsonnet/config.libsonnet (hunk 5 above)
Comment (rajaSahil) [reply jsonnet/config.libsonnet:220]: Yes, will add it.

File: jsonnet/jsonnetfile.json (hunk 6 above)
Comment (rajaSahil) [reply jsonnet/jsonnetfile.json:1]: It is the auto-generated file when we run `jb init`.

File: jsonnet/openebs-mixin/rules/prometheusRules.libsonnet (hunk 7 above)
Comment (rajaSahil) [reply jsonnet/openebs-mixin/rules/prometheusRules.libsonnet]: Changed it to `prometheus-rules`

File: jsonnet/openebs-mixin/build.sh [hunk 8]
```diff
@@ -26,7 +26,7 @@ generateRules(){
 	rm -rf $rulesDirPath
 	mkdir -p  $rulesDirPath
 	jsonnet -J vendor -m $rulesDirPath lib/rules.jsonnet
-	rulesFolder=(cStor Jiva Volume)
+	rulesFolder=(cStor Jiva Volume NPD)
```
Comment (avishnu) [inline jsonnet/openebs-mixin/build.sh]: Is there an extra space between cStor and Jiva?

File: jsonnet/openebs-mixin/rules/npd-rules.libsonnet [hunk 9]
```diff
@@ -1,8 +1,8 @@
 function(param) {
-  local rules = self,
+  //local rules = self,
```
Comment (avishnu) [inline jsonnet/openebs-mixin/rules/npd-rules.libsonnet]: Why is this commented?

File: jsonnet/openebs-mixin/rules/volume-rules.libsonnet [hunk 10]
```diff
@@ -1,12 +1,11 @@
 local volumePVC = (import 'volume-pvc-rules.libsonnet');
-local volumeNode = (import 'volume-node-rules.libsonnet');
 
 function(param) {
   local rules = self,
   _config+:: param,
   prometheusRules+:: {
     volume: {
-      [if rules._config.alertRules.volume then 'groups']+: volumePVC(rules._config).prometheusRules.volumePVC.groups + volumeNode(rules._config).prometheusRules.volumeNode.groups,
+      [if rules._config.alertRules.volume then 'groups']+: volumePVC(rules._config).prometheusRules.volumePVC.groups,
```
Comment (avishnu) [inline jsonnet/openebs-mixin/rules/volume-rules.libsonnet]: Why this change?

File: deploy/charts/openebs-monitoring/rules/cStor/cstor-rules.json [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+                  "cstorPool": "{{ $labels.cstor_pool }}",
+                  "description": "OpenEBS Pool {{ $labels.cstor_pool }} use more than 80% of his capacity",
+                  "storageClassClaim": "{{ $labels.storage_pool_claim }}",
+                  "summary": "cStor pool '{{$labels.cstor_pool}}' created for pool claim '{{$labels.storage_pool_claim}}' has {{ with printf \"openebs_free_pool_capacity{cstor_pool='%s',instance='%s',job='%s',kubernetes_pod_name='%s',slave='%s',storage_pool_claim='%s'}/(1024*1024*1024)\" $labels.cstor_pool $labels.instance $labels.job $labels.kubernetes_pod_name $labels.slave $labels.storage_pool_claim | query }} {{ . | first | value }} {{ end }}GB space remaining out of {{ with printf \"openebs_pool_size{cstor_pool='%s',instance='%s',job='%s',kubernetes_pod_name='%s',slave='%s',storage_pool_claim='%s'}/(1024*1024*1024)\" $labels.cstor_pool $labels.instance $labels.job $labels.kubernetes_pod_name $labels.slave $labels.storage_pool_claim | query }} {{ . | first | value }} {{ end }} GB. You have already consumed {{ $value }} percent of total capacity"
+               },
+               "expr": "openebs_used_pool_capacity_percent > 80",
```
Comment (avishnu) [inline deploy/charts/openebs-monitoring/rules/cStor/cstor-rules.json:15]: These numeric values should be made configurable. Can be taken up in future PR.

File: jsonnet/openebs-mixin/build.sh (hunk 8 above)
Comment (rajaSahil) [reply jsonnet/openebs-mixin/build.sh]: No, it's fine. There is only one space present.

File: jsonnet/openebs-mixin/rules/npd-rules.libsonnet (hunk 9 above)
Comment (rajaSahil) [reply jsonnet/openebs-mixin/rules/npd-rules.libsonnet]: For now, we are not using any configuration in defining the rules. That's why it is not required.

File: jsonnet/openebs-mixin/rules/volume-rules.libsonnet (hunk 10 above)
Comment (rajaSahil) [reply jsonnet/openebs-mixin/rules/volume-rules.libsonnet]: volume-node-rules is changed to npd-rules.

File: deploy/charts/openebs-monitoring/rules/cStor/cstor-rules.json (hunk 11 above)
Comment (rajaSahil) [reply deploy/charts/openebs-monitoring/rules/cStor/cstor-rules.json:15]: Sure.

File: jsonnet/README.md [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+	$ cd monitoring/jsonnet
+	```
+
+2. Generate the manifests:
+
+	```
+	$ make generate
+	```
+
+The  files in `manifests/` are the yamls that has to be applied on the kubernetes cluster.
```
Comment (avishnu) [inline jsonnet/README.md]: have* to be applied

File: jsonnet/README.md [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+
+### Apply the kube-prometheus stack and openebs monitoring addons
+The previous steps (Generate manifests) has created a bunch of manifest files in the manifests/ folder. Now simply use kubectl to install Prometheus and Grafana as per your configuration:
```
Comment (avishnu) [inline jsonnet/README.md]: Should be step-3?

File: jsonnet/config.libsonnet [hunk 14]
```diff
@@ -0,0 +1,258 @@
...
+      matchLabels: selectorName,
+    },
+    // Namespaces from which services are selected
+    namespaceSelector: {
+      any: true,
+    },
+  },
+
+  // Template function for PodMonitor
+  local podMonitor(port, path, selectorName) = {
```
Comment (avishnu) [inline jsonnet/config.libsonnet:61]: where are we using podMonitors?

Comment (rajaSahil) [reply jsonnet/config.libsonnet:61]: For cStor pool pods, we are creating a PodMonitor.

File: jsonnet/README.md (hunk 13 above)
Comment (rajaSahil) [reply jsonnet/README.md]: Yes.

File: jsonnet/openebs-mixin/README.md [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+## Generate dashboards and rules
+
+You can manually generate the alerts, dashboards and rules files. 
+
+#### Steps:
+
+1. Install `jsonnet` and `jb`
+	```
+	$ go get github.com/jsonnet-bundler/jsonnet-bundler/cmd/jb
+	$ brew install jsonnet
```
Comment (avishnu) [inline jsonnet/openebs-mixin/README.md]: How is brew installed?

File: jsonnet/openebs-mixin/README.md [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+	rulesDirPath = ''
+	```
+
+4. Finally, build the mixin:
+
+	```
+	$ make generate
+	```
+
+The  files in `rulesDirPath` are the alert rules defined for openebs which has to be used to create prometheusRule yaml then need to passed to your Prometheus server, and the files in `dashboardsDirPath` need to be imported into you Grafana server.
```
Comment (avishnu) [inline jsonnet/openebs-mixin/README.md]: need to be imported into your Prometheus.
need to be imported into your Grafana.

Comment (avishnu) [reply jsonnet/openebs-mixin/README.md]: The files (remove extra space between The and files) in `rulesDirPath` contain Prometheus-based alert rules for OpenEBS. These files will be aggregated to form a prometheusRule yaml that need to be imported into your Prometheus instance.
Similarly, the files in `dashboardsDirPath` are dashboard json files that need to be imported into your Grafana instance.

File: jsonnet/openebs-mixin/README.md [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+
+	```
+	$ make generate
+	```
+
+The  files in `rulesDirPath` are the alert rules defined for openebs which has to be used to create prometheusRule yaml then need to passed to your Prometheus server, and the files in `dashboardsDirPath` need to be imported into you Grafana server. 
+
+## Configuration
+
+The available fields and their default values can be seen in `config.libsonnet`.
```
Comment (avishnu) [inline jsonnet/openebs-mixin/README.md]: are present in

File: jsonnet/openebs-mixin/README.md [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+    dashboards: {
+      cStor: true,
+      Jiva: true,
+      LocalPV: true,
+    },
+    // AlertRules configuration. If set, then rules json will be generated.
+    alertRules: {
+      cStor: true,
+      Jiva: true,
+      lvmLocalPV: false,
```
Comment (avishnu) [inline jsonnet/openebs-mixin/README.md:57]: LVMLocalPV

File: jsonnet/openebs-mixin/README.md [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+      volume: true,
```
Comment (avishnu) [inline jsonnet/openebs-mixin/README.md:58]: How about AllVolumeTypes?

Comment (avishnu) [inline jsonnet/openebs-mixin/README.md]: Since this category `volume` contains common rules for all types of CAS engines, will it be better if we remove the switch for this and include the rules whenever any of the above CAS types are set as true?
The contention is, why would anyone want to enable cStor=true, but volume=false?

File: jsonnet/openebs-mixin/build.sh [hunk 20]
```diff
@@ -0,0 +1,40 @@
...
+generateRules(){
+	rm -rf $rulesDirPath
+	mkdir -p  $rulesDirPath
+	jsonnet -J vendor -m $rulesDirPath lib/rules.jsonnet
+	rulesFolder=(cStor Jiva Volume NPD)
+	for i in ${rulesFolder[@]}
+	do
+	    x=`echo $i | awk '{print tolower($0)}'`
+		mkdir -p $rulesDirPath/$i
+		find ./$rulesDirPath  -name "*$x*" |xargs -I{} sh -c "mv {} $rulesDirPath/$i/"
```
Comment (avishnu) [inline jsonnet/openebs-mixin/build.sh]: Is there an extra space or two before -name?

File: jsonnet/openebs-mixin/README.md (hunk 19 above)
Comment (avishnu) [reply jsonnet/openebs-mixin/README.md]: Should we specify npd as well here?

File: jsonnet/openebs-mixin/README.md (hunk 15 above)
Comment (rajaSahil) [reply jsonnet/openebs-mixin/README.md]: Have to add `sudo apt-get install jsonnet` for linux.
brew is for mac.

File: jsonnet/openebs-mixin/config.libsonnet [hunk 21]
```diff
@@ -0,0 +1,19 @@
...
+    // AlertRules configuration. If set, then rules json will be generated.
+    alertRules: {
+      cStor: true,
+      Jiva: true,
+      lvmLocalPV: false,
+      volume: true,
+      npd: true,
+    },
+  },
+}
```
Comment (avishnu) [inline jsonnet/openebs-mixin/config.libsonnet:28]: Can we make everything in lower-case to avoid any confusion?

File: jsonnet/openebs-mixin/README.md (hunk 19 above)
Comment (rajaSahil) [reply jsonnet/openebs-mixin/README.md]: Makes sense.
We can do an OR operation to check whether any of the CAS types are set as true.
 `volume: $._config.alertRules.cStor || $._config.alertRules.Jiva || $._config.alertRules.lvmLocalPV`

File: jsonnet/openebs-mixin/README.md (hunk 15 above)
Comment (avishnu) [reply jsonnet/openebs-mixin/README.md]: Add for linux.

File: jsonnet/openebs-mixin/config.libsonnet (hunk 21 above)
Comment (rajaSahil) [reply jsonnet/openebs-mixin/config.libsonnet:28]: We can but it might create confusion for lvmlocalpv or devicelocalpv.

Comment (avishnu) [reply jsonnet/openebs-mixin/config.libsonnet:28]: I don't think there will be any confusion.

File: jsonnet/openebs-mixin/README.md (hunk 19 above)
Comment (avishnu) [reply jsonnet/openebs-mixin/README.md]: Ok, make it a non-configurable variable then.

File: jsonnet/openebs-mixin/config.libsonnet (hunk 21 above)
Comment (rajaSahil) [reply jsonnet/openebs-mixin/config.libsonnet:28]: Okay. Will update it.

File: jsonnet/openebs-mixin/rules/jiva-volume-rules.libsonnet [hunk 22]
```diff
@@ -0,0 +1,30 @@
+function(param) {
```
Comment (avishnu) [inline jsonnet/openebs-mixin/rules/jiva-volume-rules.libsonnet]: Basic question: can we combine jiva-rules.libsonnet and jiva-volume-rules.libsonnet into one?

Comment (rajaSahil) [reply jsonnet/openebs-mixin/rules/jiva-volume-rules.libsonnet]: Yes, we can.

File: jsonnet/openebs-mixin/config.libsonnet [hunk 23]
```diff
@@ -1,18 +1,25 @@
...
+      cStor: $._config.casTypes.cStor,
+      jiva: $._config.casTypes.jiva,
+      localPV: $._config.casTypes.lvmLocalPV || $._config.casTypes.deviceLocalPV,
+      npd: true,
     },
     // AlertRules configuration. If set, then rules json will be generated.
     alertRules: {
-      cStor: true,
-      Jiva: true,
-      lvmLocalPV: false,
```
Comment (avishnu) [inline jsonnet/openebs-mixin/config.libsonnet]: Is this not needed?

File: jsonnet/openebs-mixin/build.sh [hunk 24]
```diff
@@ -18,20 +18,20 @@ generateDashboards(){
...
 	mkdir -p  $rulesDirPath
 	jsonnet -J vendor -m $rulesDirPath lib/rules.jsonnet
-	rulesFolder=(cStor Jiva Volume NPD)
+	rulesFolder=(cStor Jiva Volume npd)
 	for i in ${rulesFolder[@]}
 	do
 	    x=`echo $i | awk '{print tolower($0)}'`
 		mkdir -p $rulesDirPath/$i
-		find ./$rulesDirPath  -name "*$x*" |xargs -I{} sh -c "mv {} $rulesDirPath/$i/"
+		find ./$rulesDirPath -type f -name "*$x*" |xargs -I{} sh -c "mv {} $rulesDirPath/$i/"
```
Comment (avishnu) [inline jsonnet/openebs-mixin/build.sh]: Extra space at the end of line

File: jsonnet/openebs-mixin/rules/rules.libsonnet [hunk 25]
```diff
@@ -1,9 +1,25 @@
-local cstor = (import 'cstor-rules.libsonnet');
-local jiva = (import 'jiva-rules.libsonnet');
-local volume = (import 'volume-rules.libsonnet');
-local npd = (import 'npd-rules.libsonnet');
+local cstor = (import './openebs/cstor-rules.libsonnet');
+local jiva = (import './openebs/jiva-rules.libsonnet');
+local volume = (import './volume/volume-rules.libsonnet');
+local npd = (import './npd/npd-rules.libsonnet');
+
+// Alert rules for different CAS types and others(as defined in config.libsonnet under alertRules) are stored seperately in prometheusRules object.
```
Comment (avishnu) [inline jsonnet/openebs-mixin/rules/rules.libsonnet]: separately*

File: jsonnet/openebs-mixin/config.libsonnet (hunk 23 above)
Comment (rajaSahil) [reply jsonnet/openebs-mixin/config.libsonnet]: No, we don't have any alert rules for lvm-localPV.

**Review by avishnu — COMMENTED** [review]
Lgtm

File: .github/workflows/jsonnet-lint.yml [hunk 26]
```diff
@@ -0,0 +1,30 @@
...
+        with:
+          fetch-depth: 0
+
+      - name: Jsonnet formatter
+        run: |
+          cd jsonnet && make --always-make fmt
+
+      - name: Jsonnet linter
+        run: |
+          cd jsonnet && make --always-make lint
```
Comment (avishnu) [inline .github/workflows/jsonnet-lint.yml:30]: Is this tested?

Comment (rajaSahil) [reply .github/workflows/jsonnet-lint.yml:30]: Yes. https://github.com/openebs/monitoring/pull/39/checks?check_run_id=3193178661

**Review by avishnu — APPROVED** [review]
lgtm

**Review by avishnu — APPROVED** [review]

File: jsonnet/openebs-mixin/README.md (hunk 19 above)
Comment (Ab-hishek) [reply jsonnet/openebs-mixin/README.md]: This configuration section is not in sync with the actual configuration.

**Review by Ab-hishek — APPROVED** [review]
On small minor comment else everything else looks good.

File: jsonnet/openebs-mixin/dashboards/openebs/localpv-workload.json [hunk 27]
```diff
@@ -1,4216 +1,4195 @@
...
+      "1m",
+      "5m",
+      "15m",
+      "30m",
+      "1h",
+      "2h",
+      "1d"
+    ],
+    "time_options": ["5m", "15m", "1h", "6h", "12h", "24h", "2d", "7d", "30d"]
+  },
```
Comment (avishnu) [inline jsonnet/openebs-mixin/dashboards/openebs/localpv-workload.json:4190]: What are the changes exactly in this commit?

Comment (rajaSahil) [reply jsonnet/openebs-mixin/dashboards/openebs/localpv-workload.json:4190]: cStor dashboards were updated for helm charts which I forgot to update for mixin so updated the cStor dashboards and few formatting changes.
